analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -423,7 +423,10 @@ class Pip(
         // Remove the virtualenv by simply deleting the directory.
         virtualEnvDir.safeDeleteRecursively()
 
-        return ProjectAnalyzerResult(project, packages.map { it.toCuratedPackage() }.toSortedSet())
+        return ProjectAnalyzerResult(
+            project,
+            packages.map { it.toCuratedPackage() }.toSortedSet()
+        ).removePackage(Identifier("PyPI::pkg-resources:0.0.0"))
```

Comment:
Since this can only appear as part of `requirements.txt` (and not as part of `setup.py`) I'm wondering whether it'd be enough to simply filter it out similar to like we do for `python_version` at https://github.com/heremaps/oss-review-toolkit/blob/d9fdc9b6a9885a5ebd8a4e2fb19565c05ee31d14/analyzer/src/main/kotlin/managers/Pip.kt#L268

Reply:
> I'm wondering whether it'd be enough to simply filter it out similar to like we do for `python_version` at

Yes that could be done as well.
The reason I wrote it this way is that the fix is quite isolated from the existing code
and always applies. It's extremely hard to introduce a code path which bypasses the fix.
I also do not see much risk applying it always even if not needed.

After a f2f discussion with @mnonnenmacher on this review we actually decided to go into the opposite way and make the function more generic / re-usable. Maybe you can take a look at that one as well?! .. before we decide how the final solution should looks like, let me just force push it.

Reply:
I just looked at the new force-pushed version, and I have to say I still like my proposal to fix this at line 268 better. This is mostly because I find this to be quite a bit of rather complicated code being added for something that is in fact an edge case (because the bug only surfaces under very special circumstances), and it's really only being used for one single package currently.

Solving this at line 268 is much less code, and emphasizes that it's a work-around for an issue that can only appear with `requirements.txt` files.

Reply:
ok, I changed it to about what you suggested.